core/controlplane/config/templates/cloud-config-controller

```diff
@@ -2139,6 +2142,86 @@ write_files:
           - port: 80
             targetPort: 9090
 
+  - path: /srv/kubernetes/manifests/tiller.yaml
+    content: |
+        apiVersion: extensions/v1beta1
+        kind: Deployment
+        metadata:
+          creationTimestamp: null
+          labels:
+            app: helm
+            name: tiller
+          name: tiller-deploy
+          namespace: kube-system
+        spec:
+          strategy: {}
+          template:
+            metadata:
+              creationTimestamp: null
+              labels:
+                app: helm
+                name: tiller
+              # Addition to the default tiller deployment for prioritizing tiller over other non-critical pods with rescheduler
+              annotations:
+                scheduler.alpha.kubernetes.io/critical-pod: ''
+            spec:
+              tolerations:
+              # Additions to the default tiller deployment for allowing to schedule tiller onto controller nodes
+              # so that helm can be used to install pods running only on controller nodes
+              - key: "node.alpha.kubernetes.io/role"
+                operator: "Equal"
+                value: "master"
+                effect: "NoSchedule"
+              - key: "CriticalAddonsOnly"
+                operator: "Exists"
+              containers:
+              - env:
+                - name: TILLER_NAMESPACE
+                  value: kube-system
+                image: gcr.io/kubernetes-helm/tiller:v2.5.0
```

Comment:
The version number is hard-coded for now. We'd need to make this customizable like other container images.